massa-final-state/src/final_state.rs

```diff
@@ -235,12 +189,96 @@ impl FinalState {
             .expect("Cycle history should not be empty in snapshot!")
             .clone();
 
-        // Reset the cycle_history and populate it with new cycle
-        self.pos_state.cycle_history = Vec::new().into();
-        self.pos_state
-            .create_new_cycle_for_snapshot(&latest_consistent_cycle_info, end_slot);
+        if current_slot_cycle == end_slot_cycle {
+            // In that case, we just complete the gap in the same cycle
+            let latest_cycle_info =
+                self.pos_state
+                    .cycle_history
+                    .back_mut()
+                    .ok_or(FinalStateError::SnapshotError(String::from(
+                        "Invalid cycle_history",
+                    )))?;
```

Comment:
this is duplicated in the other if.  latest_cycle_info can be defined just the once, I think

Reply:
Now that it is used in two different calls, it seems a bit weird to pass around this mut (as I already have a &mut self), see refactor 1405887a97bd4fbf5444e4b09f5cf2da48919a27

Reply:
Hmmm. This feels like a dilemma. On the one hand, it is a repeated code, but on the other hand, it does seem to make sense to be in the helpers.

Might make sens to make a helper method to make the latest cycle info, but ill leave that to your discression.